### PR TITLE
Consolr dangerous asset behavior and unit tests

### PR DESCRIPTION
**Pull request by Primer42**
Fixed some bugs around how dangerous assets and statuses behave, and added unit tests.

@maddalab @byxorna @roymarantz @defect @sushruta @funzoneq

### CODE REVIEW

File: support/ruby/consolr/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+coverage/
```
Comment (roymarantz) [inline support/ruby/consolr/.gitignore]: add a new line please

Comment (Primer42) [inline support/ruby/consolr/.gitignore]: Yep, was in the  middle of fixing it when you commented.

File: support/ruby/consolr/spec/consolr_spec.rb [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    end
+
+    it 'does log list' do
+      expect { @console.start({:tag => dangerous_maintenance_tag, :log => 'list'}) }.to output("sel list\nSUCCESS\n").to_stdout_from_any_process
+    end
+
+    it 'does log clear' do
+      expect { @console.start({:tag => dangerous_maintenance_tag, :log => 'clear'}) }.to output("sel clear\nSUCCESS\n").to_stdout_from_any_process
+    end
```
Comment (roymarantz) [inline support/ruby/consolr/spec/consolr_spec.rb]: how come these log commands are considered safe and tested via that mechanism?

File: support/ruby/consolr/spec/consolr_spec.rb [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    expect { @console.start({:hostname => 'hostname-with-multiple-assets.dc.net'}) }.to raise_error(SystemExit, /Found \d+ assets/)
+  end
+
+  safe_boolean_actions = {:console => "--> Opening SOL session (type ~~. to quit)\nsol activate", :kick => 'sol deactivate', :identify => 'chassis identify', :sdr => 'sdr elist all', :on => 'power on'}
+  dangerous_boolean_actions = {:off => 'power off', :reboot => 'power cycle'}
+
+  describe 'safe allocated asset' do
+    safe_allocated_tag = 'safe-allocated-tag'
+    safe_boolean_actions.each do |action, response|
+      it "does #{action}" do
```
Comment (roymarantz) [inline support/ruby/consolr/spec/consolr_spec.rb]: perhaps adding "safe" in the it description would be clearer and symmetric with how danger is done.

**Comment by roymarantz** [comment]
This is amazing!!! and looks good to me, but reuse would be improved if:
- the initialize method took and argument, which defaulted to consolr, of the program name
- the start method was implemented something like
  def start options
      process_danger_options options  # process the danger related options and save as instance variables
     # the existing code to get the collins asset
      run asset
  end
  where #run(node) does the dangerous checks and executes the operation if it is allowed.

IMO this would allow people (me) to use this gem for other kinds of safe ipmi hacking.   e.g. nagios check...

File: support/ruby/consolr/spec/consolr_spec.rb (hunk 2 above)
Comment (Primer42) [inline support/ruby/consolr/spec/consolr_spec.rb]: `log` takes an argument (`list` or `clear`). The others take booleans.  I think we should change the way its structured, but this PR was already too big.

**Comment by Primer42** [comment]
@roymarantz - I like all those ideas, but I think I want to do them in future PRs.  This one is already pretty big, and I don't want to change too much at once.

File: support/ruby/consolr/spec/configs/collins_rspec.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+host: https://collins.dc.net
```
Comment (funzoneq) [inline support/ruby/consolr/spec/configs/collins_rspec.yml]: Best practice is to use example.com: https://tools.ietf.org/html/rfc2606#section-3

**Comment by funzoneq** [comment]
other then the nitpick :+1:

**Pull request merged**